src/perl/perl-core.c

```diff
@@ -311,10 +308,40 @@ PERL_SCRIPT_REC *perl_script_load_data(const char *data)
 /* Unload perl script */
 void perl_script_unload(PERL_SCRIPT_REC *script)
 {
+	GSList* link;
         g_return_if_fail(script != NULL);
 
 	perl_script_destroy_package(script);
-        perl_script_destroy(script);
+
+	perl_signal_remove_script(script);
+	perl_source_remove_script(script);
+
+	link = g_slist_find(perl_scripts, script);
+	if (link != NULL) {
+		perl_scripts = g_slist_remove_link(perl_scripts, link);
+		g_slist_free(link);
+		perl_script_unref(script);
+	}
+
+/*	perl_scripts = g_slist_remove(perl_scripts, script);
```

Comment:
can this commented code be removed entirely?

Reply:
Can and did.